pytensor/xtensor/shape.py

```diff
@@ -380,3 +382,75 @@ def squeeze(x, dim=None):
         return x  # no-op if nothing to squeeze
 
     return Squeeze(dims=dims)(x)
+
+
+class ExpandDims(XOp):
+    """Add a new dimension to an XTensorVariable."""
+
+    __props__ = ("dims",)
+
+    def __init__(self, dim):
+        self.dims = dim
```

Comment:
Nit: If the Op only allows one dim use singular, if it allows multiple use plural name. 

Important: Do not create `__init__` with a different signature than the variable name you end up saving into.

Reply:
Also feel free to do any validation here, must dim be a string? Check for that

Reply:
I see now that `xarray.DataArray.expand_dims` allows the `dim` parameter to be "hashable, sequence of hashable, dict, or None "

Currently we are handling only string and None.

Should we generalize to handle all cases handled by xarray?